### PR TITLE
Updating variables in Celldet

### PR DESCRIPTION
**Pull request by Ishita-S**
## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

File: Exec/science/celldet/problem_initialize.H [hunk 1]
```diff
@@ -64,17 +155,48 @@ void problem_initialize ()
...
-    eos_state.T = problem::T_r;
+    eos_state.T = problem::T_fuel;
 
     eos(eos_input_rt, eos_state);
 
-    problem::ambient_e_r = eos_state.e;
+    problem::ambient_e_fuel = eos_state.e;
+
+    eos_t eos_state;
+    eos_state.rho = problem::rho_fuel;
```
Comment (harpolea) [inline Exec/science/celldet/problem_initialize.H:171]:
```suggestion
    eos_state.rho = problem::dens_fuel;
```

File: Exec/science/celldet/problem_initialize.H [hunk 2]
```diff
@@ -64,17 +155,48 @@ void problem_initialize ()
...
+
+    eos(eos_input_rt, eos_state);
+
+    problem::e_fuel = eos_state.e;
+    problem::p_fuel = eos_state.p;
+
+    // compute the ash state -- this should be hot but in pressure
+    // equilibrium
+
+    eos_state.rho = problem::rho_fuel;  // initial guess
```
Comment (harpolea) [inline Exec/science/celldet/problem_initialize.H:185]:
```suggestion
    eos_state.rho = problem::dens_fuel;  // initial guess
```

File: Exec/science/celldet/problem_initialize.H [hunk 3]
```diff
@@ -64,17 +155,48 @@ void problem_initialize ()
...
+    }
+
+    eos(eos_input_tp, eos_state);
+
+    problem::rho_ash = eos_state.rho;
+    problem::e_ash = eos_state.e;
+
+    // mass flux will be constant across the flame
+
+    problem::mass_flux = problem::rho_fuel * problem::v_inflow;
```
Comment (harpolea) [inline Exec/science/celldet/problem_initialize.H:199]:
```suggestion
```
We don't need the mass flux for this problem

File: Exec/science/celldet/problem_initialize.H [hunk 4]
```diff
@@ -64,17 +155,48 @@ void problem_initialize ()
...
+    eos_state.rho = problem::rho_fuel;  // initial guess
+    eos_state.p = problem::p_fuel;      // pressure equilibrum
+    eos_state.T = problem::T_ash;
+    for (int n = 0; n < NumSpec; n++) {
+        eos_state.xn[n] = problem::xn_ash[n];
+    }
+
+    eos(eos_input_tp, eos_state);
+
+    problem::rho_ash = eos_state.rho;
```
Comment (harpolea) [inline Exec/science/celldet/problem_initialize.H:194]:
```suggestion
    problem::dens_ash = eos_state.rho;
```

File: Exec/science/celldet/problem_initialize.H [hunk 5]
```diff
@@ -12,6 +12,97 @@ AMREX_INLINE
...
+    for (int n = 0; n < NumSpec; n++) {
+        sum_Xf += problem::xn_fuel[n];
+        sum_Xa += problem::xn_ash[n];
+    }
+    for (int n = 0; n < NumSpec; n++) {
+        problem::xn_fuel[n] /= sum_Xf;
+        problem::xn_ash[n] /= sum_Xa;
+    }
+
     // get the species indices
```
Comment (harpolea) [inline Exec/science/celldet/problem_initialize.H:106]: I'm not convinced you actually need any of the stuff from this point onwards

File: Exec/science/celldet/problem_initialize_state_data.H [hunk 6]
```diff
@@ -30,9 +30,13 @@ void problem_initialize_state_data (int i, int j, int k,
...
+    state(i,j,k,URHO) = problem::dens_ash + (problem::dens_fuel - problem::dens_ash) * (1.0_rt - sigma);
 
-    state(i,j,k,UTEMP) = problem::T_l + (problem::T_r - problem::T_l) * (1.0_rt - sigma);
+    state(i,j,k,UTEMP) = problem::T_ash + (problem::T_fuel - problem::T_ash) * (1.0_rt - sigma);
+
+    for (int n = 0; n < NumSpec; n++) {
+        problem::xn[n] = problem::xn_ash[n] + (problem::xn_fuel[n] - problem::xn_ash[n]) * (1.0_rt - sigma);
+}
 
     for (int n = 0; n < NumSpec; n++) {
```
Comment (harpolea) [inline Exec/science/celldet/problem_initialize_state_data.H:41]:
```suggestion
```
You don't need to have these in separate loops

**Review by harpolea — CHANGES_REQUESTED** [review]

File: Exec/science/celldet/inputs.celldet [hunk 7]
```diff
@@ -71,19 +71,23 @@ amr.derive_plot_vars = ALL
 
 #PROBIN FILENAME
 
-problem.T_l = 4.e9
-problem.T_r = 5.e7
+problem.T_l = 4.423e9
+problem.T_r = 2.e8
```
Comment (zingale) [inline Exec/science/celldet/inputs.celldet:75]: you renamed these to T_ash and T_fuel, and similarly rho_ash and rho_fuel, so you need to update the inputs

File: Exec/science/celldet/_prob_params [hunk 8]
```diff
@@ -1,10 +1,10 @@
...
-T_r                  real          5.e7_rt        y
+T_fuel               real          5.e7_rt        y
 
-dens_l               real          1.e8_rt        y
+dens_ash             real          1.e8_rt        y
 
-dens_r               real          1.e8_rt        y
+dens_fuel            real          1.e8_rt        y
 
 cfrac                real          0.5_rt         y
```
Comment (zingale) [inline Exec/science/celldet/_prob_params:9]: is cfrac used still?

File: Exec/science/celldet/inputs.celldet [hunk 9]
```diff
@@ -87,6 +86,10 @@ problem.center_T = 1.e-1
 
 problem.vel = 2.876e8
 
+<<<<<<< HEAD:Exec/science/celldet/inputs.celldet
+
+=======
+>>>>>>> 7bcddcc23b7c579f737c5c7feb713ec90bb77469:Exec/science/celldet/inputs.celldet
```
Comment (zingale) [inline Exec/science/celldet/inputs.celldet:93]: can you remove these merge conflict lines?

File: Exec/science/celldet/nse19.tbl [hunk 10]
```diff
@@ -0,0 +1 @@
+../../../external/Microphysics/networks/aprox19/nse19.tbl
```
Comment (zingale) [inline Exec/science/celldet/nse19.tbl:1]: this file should not be part of the PR

File: Exec/science/celldet/problem_initialize_state_data.H [hunk 11]
```diff
@@ -30,19 +30,30 @@ void problem_initialize_state_data (int i, int j, int k,
...
+<<<<<<< HEAD
+        xn[n] = problem::xn_ash[n] + (problem::xn_fuel[n] - problem::xn_ash[n]) * (1.0_rt - sigma);
+}
 
     for (int n = 0; n < NumSpec; n++) {
+        state(i,j,k,UFS+n) = state(i,j,k,URHO) * xn[n];
+=======
+        problem::xn[n] = problem::xn_ash[n] + (problem::xn_fuel[n] - problem::xn_ash[n]) * (1.0_rt - sigma);
         state(i,j,k,UFS+n) = state(i,j,k,URHO) * problem::xn[n];
+>>>>>>> cff4ab0ffa9e10c81e2cbec9a2be8d68e661b136
```
Comment (zingale) [inline Exec/science/celldet/problem_initialize_state_data.H:49]: you need to remove / resolve this merge conflict

File: Exec/science/celldet/species_diag.out [hunk 12]
```diff
@@ -0,0 +1,339 @@
+#   COLUMN 1                        2                        3                        4                        5                        6                        7                        8                        9                       10                       11                       12                       13                       14                       15                       16                       17                       18                       19                       20                       21
```
Comment (zingale) [inline Exec/science/celldet/species_diag.out:1]: this file should not be part of the PR

**Comment by zingale** [comment]
can you merge the latest `development` into the PR?

File: Exec/science/celldet/problem_initialize_state_data.H [hunk 13]
```diff
@@ -30,19 +30,28 @@ void problem_initialize_state_data (int i, int j, int k,
...
+
+    Real xn[NumSpec] ;
+
+    for (int n = 0; n < NumSpec; n++) {
+        xn[n] = problem::xn_ash[n] + (problem::xn_fuel[n] - problem::xn_ash[n]) * (1.0_rt - sigma);
+}
 
     for (int n = 0; n < NumSpec; n++) {
+        state(i,j,k,UFS+n) = state(i,j,k,URHO) * xn[n];
+=======
```
Comment (zingale) [inline Exec/science/celldet/problem_initialize_state_data.H:45]: this is left behind from a merge

File: Exec/science/celldet/amr_diag.out [hunk 14]
```diff
@@ -0,0 +1,341 @@
+#   COLUMN 1                        2                        3           4                        5
```
Comment (zingale) [inline Exec/science/celldet/amr_diag.out:1]: this file should not be part of the PR

File: Exec/science/celldet/grid_diag.out [hunk 15]
```diff
@@ -0,0 +1,338 @@
+                     time                     mass                     xmom                     ymom                     zmom                ang mom x                ang mom y                ang mom z                    rho_K                    rho_e                    rho_E
```
Comment (zingale) [inline Exec/science/celldet/grid_diag.out:1]: this file should not be part of the PR

**Comment by zingale** [comment]
I think you might need to pull in the submodules?  I am not sure why it wants to update amrex and Microphysics, but those should not be part of this PR

**Comment by Ishita-S** [comment]
Does the most recent commit I made resolve the issue?

**Comment by zingale** [comment]
it looks like the files are still there.

**Comment by Ishita-S** [comment]
I manually deleted them one GitHub so hopefully that clears that issue

**Pull request closed without merging**